airbyte-integrations/connectors/source-pinterest/source_pinterest/components/record_filter.py

```diff
@@ -0,0 +1,41 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, List, Mapping, Optional
+
+import pendulum
+from airbyte_cdk.sources.declarative.extractors.record_filter import RecordFilter
+from airbyte_cdk.sources.declarative.types import StreamSlice, StreamState
+
+
+class PinterestSemiIncrementalRecordFilter(RecordFilter):
+    """
+    This class extends the RecordFilter functionality
+    and allows to filter by state or start date semi-incremental substreams
+    """
+
+    cursor_field = "updated_time"
+
+    def filter_records(
+        self,
+        records: List[Mapping[str, Any]],
+        stream_state: StreamState,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> List[Mapping[str, Any]]:
+        cursor_value = max(self._get_config_start_date(), self._get_current_partition_state(stream_state, stream_slice))
+        return [record for record in records if record[self.cursor_field] >= cursor_value]
+
+    def _get_config_start_date(self) -> int:
+        start_date = self.config.get("start_date")
+        return int(pendulum.parse(start_date).timestamp()) if start_date else 0
+
+    def _get_current_partition_state(self, stream_state: StreamState, stream_slice: Optional[StreamSlice] = None) -> int:
```

Comment:
From what I can see, the main purpose of this method is to retrieve the state for the current partition. I believe this logic has already been implemented, and you can easily obtain the required data in low-code just like that:
```
record_filter:
    condition: "{{ record['updated_time'] > stream_slice['start_time'] }}"
```
Using this, I think we can avoid using a custom component. If the state is present, `stream_slice['start_time']` will provide you with the current state value; otherwise, it will be the date from the configuration.

Reply:
This has been updated. Thanks for the advice!